docfx/articles/memory.md

```diff
@@ -140,6 +140,69 @@ Rather than passing tensor arguments between neural network layers inside a cust
 
 It may not be ideal when first experimenting with a model and trying to debug it, but once you are done with that and move on to a full training data set, it is advisable.
 
+## Technique 3: torch.NewDisposeScope()
+
+This approach, which was added in TorchSharp 0.95.4, makes it easier to dispose of tensors, without the non-determinism of technique 1, or the many temporaries of technique 2. It has most of the advantages of technique 2, and the code elegance of technique 1.
+
+👍 Specific lifetime management of all resources, but in groups.
+
+👍 Temporaries __are__ covered by this approach.
+
+👎 You don't have fine-grained control over when each tensor is reclaimed. All tensors created while a scope is in effect are disposed at once.
+
+👎 It's a new code pattern, not widely used with other libraries.
+
+Let's look at an example:
+
+```C#
+using (var d = torch.NewDisposeScope()) {
+
+    total_acc += (predicted_labels.argmax(1) == labels).sum().cpu().item<long>();
```

Comment:
I am curious about this example. In C# `total_acc += x` is the same as `total_acc = total_acc + x`, which in TorchSharp will:
- create a new instance of `Tensor` for `total_acc + x`
- assign it to `total_acc` rendering the old `Tensor` previously stored in the var unreachable
- from my understanding of `NewDisposeScope` leaving the scope of `using` will dispose the new `Tensor` that is now stored in `total_acc`

Reply:
Good catch. Yes, it has to be excluded from disposal. The example is too simplistic.

Reply:
Of course, in this particular example, total_acc is an integer, not a tensor...

Reply:
Ooops, did not notice the last op was data access.